code/controllers/failsafe.dm

```diff
@@ -1,102 +1,102 @@
-
-/datum/controller/failsafe/Destroy()
-	running = FALSE
-	..()
-	return QDEL_HINT_HARDDEL_NOW
-
-/datum/controller/failsafe/proc/Loop()
-	while(running)
-		lasttick = world.time
-		if(!Master)
-			// Replace the missing Master! This should never, ever happen.
-			new /datum/controller/master()
-		// Only poke it if overrides are not in effect.
-		if(processing_interval > 0)
-			if(Master.processing && Master.iteration)
-				// Check if processing is done yet.
-				if(Master.iteration == master_iteration)
-					switch(defcon)
-						if(4,5)
-							--defcon
-						if(3)
-							message_admins("Notice: DEFCON [defcon_pretty()]. The Master Controller has not fired in the last [(5 - defcon) * processing_interval] ticks.")
-							--defcon
-						if(2)
-							to_chat(GLOB.admins, "<span class='boldannounce'>Warning: DEFCON [defcon_pretty()]. The Master Controller has not fired in the last [(5-defcon) * processing_interval] ticks. Automatic restart in [processing_interval] ticks.</span>")
-							--defcon
-						if(1)
-
-							to_chat(GLOB.admins, "<span class='boldannounce'>Warning: DEFCON [defcon_pretty()]. The Master Controller has still not fired within the last [(5-defcon) * processing_interval] ticks. Killing and restarting...</span>")
-							--defcon
-							var/rtn = Recreate_MC()
-							if(rtn > 0)
-								defcon = 4
-								master_iteration = 0
-								to_chat(GLOB.admins, "<span class='adminnotice'>MC restarted successfully</span>")
-							else if(rtn < 0)
-								WARNING("FailSafe: Could not restart MC, runtime encountered. Entering defcon 0")
-								to_chat(GLOB.admins, "<span class='boldannounce'>ERROR: DEFCON [defcon_pretty()]. Could not restart MC, runtime encountered. I will silently keep retrying.</span>")
-							//if the return number was 0, it just means the mc was restarted too recently, and it just needs some time before we try again
-							//no need to handle that specially when defcon 0 can handle it
-						if(0) //DEFCON 0! (mc failed to restart)
-							var/rtn = Recreate_MC()
-							if(rtn > 0)
-								defcon = 4
-								master_iteration = 0
-								to_chat(GLOB.admins, "<span class='adminnotice'>MC restarted successfully</span>")
-				else
-					defcon = min(defcon + 1,5)
-					master_iteration = Master.iteration
-			if (defcon <= 1)
-				sleep(processing_interval*2)
-			else
-				sleep(processing_interval)
-		else
-			defcon = 5
-			sleep(initial(processing_interval))
-
-/datum/controller/failsafe/proc/defcon_pretty()
-	return defcon
-
-/datum/controller/failsafe/stat_entry()
-	if(!statclick)
-		statclick = new/obj/effect/statclick/debug(null, "Initializing...", src)
-
-	stat("Failsafe Controller:", statclick.update("Defcon: [defcon_pretty()] (Interval: [Failsafe.processing_interval] | Iteration: [Failsafe.master_iteration])"))
+/**
+* Failsafe
+*
+* Pretty much pokes the MC to make sure it's still alive.
+**/
+
+GLOBAL_REAL(Failsafe, /datum/controller/failsafe)
```

Comment:
Think you could do this again without the CRLF/LF changes?

Reply:
Sure I'll try